src/erlcloud_ec2.erl

```diff
@@ -1941,46 +2051,187 @@ describe_snapshot_attribute(SnapshotID, create_volume_permission, Config)
             Error
     end.
 
-%%
-%%
--spec(describe_snapshots/0 :: () -> [proplist()]).
-describe_snapshots() -> describe_snapshots([], "self").
+-type ec2_snapshot_ids() :: [string()].
+-type ec2_snapshot_owner() :: string().
+-type ec2_snapshot_restorable_by() :: string() | none.
 
--spec(describe_snapshots/1 :: ([string()] | aws_config()) -> proplist()).
-describe_snapshots(Config)
-  when is_record(Config, aws_config) ->
-    describe_snapshots([], "self", Config);
-describe_snapshots(SnapshotIDs) ->
-    describe_snapshots(SnapshotIDs, none, none, default_config()).
+-define(SNAPSHOTS_MR_MIN, 5).
+-define(SNAPSHOTS_MR_MAX, 1000).
+-define(SNAPSHOTS_MR_DEFAULT, 1000).
```

Comment:
Some functions have different max and min values, I'll see if I can optimise the defines once I've implemented all 7 functions